src/pages/lessons/[slug].tsx

```diff
@@ -155,19 +155,41 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
   const actualPlayerRef = React.useRef<any>(null)
   const lastAutoPlayed = React.useRef()
 
+  const [watchCount, setWatchCount] = React.useState<number>(0)
+
   const [playerState, send] = useMachine(playerMachine, {
     context: {
       lesson: initialLesson,
       viewer,
     },
+    services: {
+      loadLesson: async () => {
+        if (cookieUtil.get(`egghead-watch-count`)) {
+          setWatchCount(Number(cookieUtil.get(`egghead-watch-count`)))
+        } else {
+          setWatchCount(
+            Number(
+              cookieUtil.set(`egghead-watch-count`, 0, {
+                expires: 15,
+              }),
+            ),
+          )
+        }
+
+        console.debug('loading video with auth')
+        const loadedLesson = await loadLesson(initialLesson.slug)
+        console.debug('authed video loaded', {video: loadedLesson})
+
+        return loadedLesson
+      },
+    },
```

Comment:
The implementation of `loadLesson` -- this moves the logic from a couple of useEffects into this single spot that the machine can invoked internally.